pandas/core/array_algos/categorical_accumulations.py

```diff
@@ -0,0 +1,60 @@
+"""
+categorical_accumulations.py is for accumulation algorithms using a mask-based
+approach for missing values.
+"""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+
+
+def _cum_func(
```

Comment:
Can you just include all this logic in `pandas/core/arrays/categorical.py`

Reply:
I moved the logic into `pandas/core/arrays/categorical.py`